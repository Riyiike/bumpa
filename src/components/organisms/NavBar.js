import React, { memo, useContext, useState } from "react";
import moonIcon from "../../assets/moonIcon.svg";
import sunIcon from "../../assets/sunIcon.svg";
import { Link } from "react-router-dom";
import baseUrl from "../../utils/route/baseUrl";
import NavBarContext from "../../contexts/NavBarContext";

const NavBar = () => {
  const { darkTheme, setDarkTheme } = useContext(NavBarContext);

  function toggleTheme() {
    setDarkTheme((oldValue) => !oldValue);
  }

  return (
    <>
      <nav className="bg-White dark:bg-DarkBlue drop-shadow-md md:relative ">
        <div className="max-w-7xl mx-auto flex justify-between items-center px-4 py-6">
          <Link to={`${baseUrl}`}>
            <h1 className=" font-semibold text-lg md:text-xl ">
              Where is the world ?
            </h1>
          </Link>

          {/* nav buttons */}
          <button onClick={toggleTheme} className="flex gap-1 items-center">
            <img
              src={darkTheme ? sunIcon : moonIcon}
              alt="theme"
              className="w-4 dark:w-5 dark:invert"
            />
            {darkTheme ? "Light" : "Dark"} Mode
          </button>
        </div>
      </nav>
    </>
  );
};

export default NavBar;
