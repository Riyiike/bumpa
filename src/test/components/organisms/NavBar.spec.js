import React from "react";
import { shallow } from 'enzyme'
import NavBar from "../../../components/organisms/NavBar";


jest.mock('react', () => ({
    ...jest.requireActual('react'),
    useContext: () => ({
        darkTheme: "NavBarContext",
    }),

}))


describe('test mainPage', () => {
    it('test mainPage', () => {
        const btn = shallow(<NavBar />)
        expect(btn).toMatchSnapshot();
    })
})