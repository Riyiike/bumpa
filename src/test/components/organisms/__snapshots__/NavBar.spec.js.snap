// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test mainPage test mainPage 1`] = `
<Fragment>
  <nav
    className="bg-White dark:bg-DarkBlue drop-shadow-md md:relative "
  >
    <div
      className="max-w-7xl mx-auto flex justify-between items-center px-4 py-6"
    >
      <Link
        to="/"
      >
        <h1
          className=" font-semibold text-lg md:text-xl "
        >
          Where is the world ?
        </h1>
      </Link>
      <button
        className="flex gap-1 items-center"
        onClick={[Function]}
      >
        <img
          alt="theme"
          className="w-4 dark:w-5 dark:invert"
          src="sunIcon.svg"
        />
        Light
         Mode
      </button>
    </div>
  </nav>
</Fragment>
`;
